weblate/trans/models/component.py

```diff
@@ -1179,6 +1211,106 @@ def get_repoweb_link(
             component=self,
         )
 
+    def get_git_repoweb_template(self):
+        """Method to return the template link for a specific vcs."""
+        parsed_url = urlparse(self.repo)
+
+        if (
+            self.repo.startswith("git@bitbucket.org")
+            or parsed_url.hostname == "bitbucket.org"
+        ):
+            return self.get_bitbucket_git_repoweb_template()
+
+        if (
+            self.repo.startswith("git@github.com")
+            or parsed_url.hostname == "github.com"
+        ):
+            return self.get_github_repoweb_template()
+
+        if parsed_url.hostname == "pagure.io":
+            return self.get_pagure_repoweb_template()
+
+        if (
+            self.repo.startswith("git@ssh.dev.azure.com:v3")
+            or parsed_url.hostname == "dev.azure.com"
+            or parsed_url.hostname.split(".", 1)[-1] == "visualstudio.com"
+        ):
+            return self.get_azure_repoweb_template()
+
+        return None
+
+    def get_clean_slug(self, slug):
+        if slug.endswith(".git"):
+            slug = slug.split(".git")[0]
```

Comment:
This will break when somebody names repo for example `some.gitorious.git`.

Better to just strip last four chars when we already know them:

```suggestion
            slug = slug[:-4]
```

Alternatively, the suffix can be stripped in the regexp while parsing the URL.

Reply:
I think regexp stripping is safer than stripping. I will do that. Thanks